scripts/check_urls/check_urls.ts

```diff
@@ -0,0 +1,57 @@
+import fs from "fs";
+import YAML from "yaml";
+
+interface UrlObject {
+  description: string;
+  tags: string[];
+  title: string;
+  url: string;
+}
+const filePath = "../../data/resources.yml";
+
+fs.readFile(filePath, "utf-8", (err, data) => {
+  const urls: UrlObject[] = YAML.parse(data);
+
+  const checkUrl = async (url: UrlObject) => {
+    try {
+      const response = await fetch(new URL(url.url).toString(), {
+        method: "HEAD",
+      });
+      if (response.status == 200 && !response.redirected) {
+        return { url: url.url, status: response.status };
+      } else if (response.redirected) {
+        return {
+          url: url.url,
+          status: 301,
+          redirect: response.url,
+        };
+      } else {
+        return { url: url.url, status: response.status };
+      }
+    } catch (err) {
+      console.error(err);
+      return { url: url.url, status: "Error" };
+    }
+  };
+
+  Promise.all(urls.map(checkUrl)).then((links) => {
+    links
+      .filter((l) => l.status == 200)
+      .forEach((link) => {
+        console.log(`${link.url} - OK ${link.status}`);
+      });
+
+    const errorLinks = links.filter((l) => l.status != 200);
```

Comment:
Hmm. This is interesting. `301` codes are only problematic if they direct to a different domain (e.g. `www.350.org` redirecting to `350.org` doesn't matter), as that indicates that they are switching domains and the old one may become a `404` error sometime. Would you be able to implement a check for this?

Reply:
@VeckoTheGecko I think the new code should resolve it. I do a check where if the domain is the same then it passes as a live link.